src/main/scala/pt/tecnico/dsi/openstack/keystone/services/Roles.scala

```diff
@@ -55,14 +54,44 @@ final class Roles[F[_]: Sync: Client](baseUri: Uri, authToken: Header)
   }
   
   /** Allows performing role assignment operations on the domain with `id` */
-  def onDomain(id: String): RoleAssignment[F] = new RoleAssignment(baseUri / "domains" / id, authToken)
+  def onDomain(id: String): RoleAssignment[F] =
+    new RoleAssignment(baseUri, Scope.Domain.id(id), authToken)
   /** Allows performing role assignment operations on the project with `id` */
-  def onProject(id: String): RoleAssignment[F] = new RoleAssignment(baseUri / "projects" / id, authToken)
+  def onProject(id: String): RoleAssignment[F] =
+    new RoleAssignment(baseUri, Scope.Project(id), authToken)
   
   /** Allows performing role assignment operations on `domain`. */
   def on(domain: Domain): RoleAssignment[F] = onDomain(domain.id)
   /** Allows performing role assignment operations on `project`. */
   def on(project: Project): RoleAssignment[F] = onDomain(project.id)
   /** Allows performing role assignment operations on system. */
-  def on(@nowarn system: System.type): RoleAssignment[F] = new RoleAssignment(baseUri / "system", authToken)
+  def on(@nowarn system: System.type): RoleAssignment[F] =
+    new RoleAssignment(baseUri, Scope.System(), authToken)
+  
+  def listAssignments(userId: Option[String] = None, groupId: Option[String] = None, roleId: Option[String] = None,
+    domainId: Option[String] = None, projectId: Option[String] = None, system: Option[Boolean] = None,
+    effective: Boolean = false, includeNames: Boolean = false, includeSubtree: Boolean = false): Stream[F, Assignment] = {
+    val query: Vector[(String, Option[String])] = Vector(
+      "user.id" -> userId,
+      "group.id" -> groupId,
+      "role.id" -> roleId,
+      "scope.domain.id" -> domainId,
+      "scope.project.id" -> projectId,
+      "scope.system" -> system.map(_.toString),
+    ).filter { case (_, value) => value.isDefined } ++
```

Comment:
Codacy found an issue: [Usage of isDefined on optional type.](https://app.codacy.com/gh/ist-dsi/scala-keystone-client/pullRequest?prid=6111885)